src/main/scala/ai/privado/languageEngine/python/passes/config/PythonPropertyFilePass.scala

```diff
@@ -0,0 +1,179 @@
+
+class PythonPropertyFilePass(cpg: Cpg, projectRoot: String) extends ForkJoinParallelCpgPass[String](cpg) {
+  override def generateParts(): Array[String] = {
+    configFiles(projectRoot, Set(".ini", ".yml", ".yaml", ".env")).toArray
+  }
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  override def runOnPart(builder: DiffGraphBuilder, file: String): Unit = {
+    val fileNode      = addFileNode(file, builder)
+    val propertyNodes = addPropertyNodesAndConnectToUsers(file, builder)
+    propertyNodes.foreach(builder.addEdge(_, fileNode, EdgeTypes.SOURCE_FILE))
+  }
+
+  private def addPropertyNodesAndConnectToUsers(
+    file: String,
+    builder: BatchedUpdate.DiffGraphBuilder
+  ): List[NewJavaProperty] = {
+    Try {
+      obtainKeyValuePairs(file).filter(pair => pair._1.nonEmpty && pair._2.nonEmpty)
+    } match {
+      case Success(keyValuePairs) =>
+        val propertyNodes = keyValuePairs.map(addPropertyNode(_, builder))
+
+        propertyNodes.foreach(propertyNode => {
+          connectGetEnvironLiterals(propertyNode, builder)
+          connectDBConfigMembers(propertyNode, builder)
+        })
+
+        propertyNodes
+      case Failure(exception) =>
+        logger.warn(exception.getMessage)
+        List()
+    }
+  }
+
+  private def obtainKeyValuePairs(file: String): List[(String, String)] = {
+    if (file.endsWith(".conf")) {
+      parseConfFiles(file)
+    } else if (file.endsWith(".ini")) {
+      parseINIFiles(file)
+    } else {
+      getDotenvKeyValuePairs(file)
+    }
+  }
+
+  private def getDotenvKeyValuePairs(file: String): List[(String, String)] = {
+    val envProps = new Properties()
+    Source
+      .fromFile(file)
+      .getLines()
+      .filter(line => line.trim.nonEmpty && !line.startsWith("#"))
+      .foreach(line => {
+        val Array(key, value) = line.split("=", 2);
+        envProps.setProperty(key, value)
+      })
+
+    envProps.asScala
+      .map(prop => (prop._1, prop._2))
+      .toList
+  }
+
+  private def matchEnvironGetCalls(propertyName: String): List[Literal] = {
+    cpg.literal
+      .codeExact("\"" + propertyName + "\"")
+      .where(_.inCall.methodFullName(".*\\(?environ\\)?\\.get"))
+      .l
+  }
+
+  private def configFiles(projectRoot: String, extensions: Set[String]): List[String] = {
+
+    def getListOfFiles(dir: String): List[File] = {
+      val d = new File(dir)
+      if (d.exists && d.isDirectory) {
+        d.listFiles.filter(_.isFile).toList
+      } else {
+        List[File]()
+      }
+    }
+
+    val configFileList = SourceFiles
+      .determine(Set(projectRoot), extensions)
+      .filter(Utilities.isFileProcessable)
+      .concat(getListOfFiles(projectRoot).map(f => f.getAbsolutePath).filter(_.matches(".*\\.env.*")))
+
+    SourceFiles
+      .determine(Set(projectRoot), extensions)
+      .filter(Utilities.isFileProcessable)
+      .concat(getListOfFiles(projectRoot).map(f => f.getAbsolutePath).filter(_.matches(".*\\.env.*")))
```

Comment:
Why do we need concat here?

Reply:
And this same piece of code is present above and assigned to configFileList. Any reason?

Reply:
We don't just have files with extension .env, but also files such as .env.development or .env.local. We'll need to track those as well.

Reply:
Ok, then also apply the filter(Utilities.isFileProcessable) on this.

But still wondering why the same code is present twice, on Line no 99 and 104?

Reply:
Sorry, missed it. It was for debugging.